src/script/conversation/ConversationAccessPermission.ts

```diff
@@ -0,0 +1,150 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import {CONVERSATION_ACCESS, ACCESS_ROLE_V2} from '@wireapp/api-client/src/conversation/';
+import {combinePermissions, hasPermissions} from '../user/UserPermission';
+import {ACCESS_STATE, TEAM} from './AccessState';
+
+export const ACCESS_TYPES = {
+  GUEST: 1 << 0,
+  NON_TEAM_MEMBER: 1 << 1,
+  TEAM_MEMBER: 1 << 2,
+  SERVICE: 1 << 3,
+};
+
+export const ACCESS_MODES = {
+  INVITE: 1 << 4,
+  CODE: 1 << 5,
+};
+
+export const ACCESS = {...ACCESS_TYPES, ...ACCESS_MODES};
+
+export function teamPermissionsForAccessState(state: ACCESS_STATE): number {
+  switch (state) {
+    case ACCESS_STATE.TEAM.GUESTS_SERVICES: {
+      return combinePermissions([teamPermissionsForAccessState(ACCESS_STATE.TEAM.GUEST_ROOM), ACCESS_TYPES.SERVICE]);
+    }
+    case ACCESS_STATE.TEAM.GUEST_ROOM: {
+      return combinePermissions([
+        teamPermissionsForAccessState(ACCESS_STATE.TEAM.TEAM_ONLY),
+        teamPermissionsForAccessState(ACCESS_STATE.TEAM.GUEST_FEATURES),
+      ]);
+    }
+    case ACCESS_STATE.TEAM.GUEST_FEATURES: {
+      return combinePermissions([ACCESS_TYPES.GUEST, ACCESS_MODES.CODE, ACCESS_TYPES.NON_TEAM_MEMBER]);
+    }
+    case ACCESS_STATE.TEAM.SERVICES: {
+      return combinePermissions([teamPermissionsForAccessState(ACCESS_STATE.TEAM.TEAM_ONLY), ACCESS_TYPES.SERVICE]);
+    }
+    case ACCESS_STATE.TEAM.TEAM_ONLY: {
+      return combinePermissions([ACCESS_MODES.INVITE, ACCESS_TYPES.TEAM_MEMBER]);
+    }
+    default: {
+      return 0;
+    }
+  }
+}
+
+export function hasAccessToFeature(feature: number, state: ACCESS_STATE): boolean {
+  const permissions = teamPermissionsForAccessState(state);
+  return !!(feature & permissions);
+}
+
+export function isGettingAccessToFeature(feature: number, prevState: ACCESS_STATE, current: ACCESS_STATE) {
+  return !hasAccessToFeature(feature, prevState) && hasAccessToFeature(feature, current);
+}
+
+export function featureFromStateChange(prevState: ACCESS_STATE, current: ACCESS_STATE) {
+  if (prevState === current) {
+    return {feature: undefined, featureName: undefined, isAvailable: undefined, number: 0};
+  }
+  const [featureName, featureValue] = Object.entries(ACCESS).find(
+    ([, value]) => value & (teamPermissionsForAccessState(prevState) ^ teamPermissionsForAccessState(current)),
+  );
+  const featString = ACCESS_ROLE_V2[featureName as keyof typeof ACCESS_ROLE_V2];
+  return {
+    feature: featString,
+    featureName: (featString?.[0].toUpperCase() + featString?.slice(1)) as 'Guest' | 'Service',
+    isAvailable: hasAccessToFeature(featureValue, current),
+    number: featureValue,
```

Comment:
I think the naming here is a little confusing. Would `bitmask` work for you rather than `number`?